src/regex.nim

```diff
@@ -2100,6 +2101,51 @@ proc groupsCount*(m: RegexMatch): int =
   ##
   m.groups.len
 
+
+proc groupsNames*(m: RegexMatch): seq[string] =
+  ## return the names of capturing groups
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupsCount == 2
+
+
+  toSeq(m.namedGroups.keys)
+
+
+proc groupsByName*(m: RegexMatch, name: string, originalText: string): seq[string] =
+  ## return capturing groups by name
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupByName("greet") == @["hello"]
+  ##     doAssert m.groupByName("who") == @["world"]
+  
+  for bounds in m.group(name):
+    result.add(originalText[bounds])
+
+
+proc groupByName*(m: RegexMatch, name: string, originalText: string): string =
+  ## return capturing groups by name
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupByName("greet") == "hello"
+  ##     doAssert m.groupByName("who") == "world"
```

Comment:
Rename it to `groupFirstCapture`, and rename the `name` param to `groupName`. Description should say something like `Return first capture for a given capturing group`.

mmh, PCRE and most other regex engines return the last capture, not the first. However, first is fine by me, as long as the proc name makes it obvious.

Reply:
Apparently the feature isn't available it always returning the last capturing group so I removed it all together and left it as groupCapture

Reply:
I'm not asking to change it to last capture. I can do that later. However, `groupCapture` does not tell me whether I'm getting the first or the last capture, and I think getting the first would be more surprising than getting the last capture, given most regex engine seem to return the last one.

Reply:
I meant in your code whenever I try to get all of the groups by name it always returns the last one, that's why I changed it.

Reply:
Oh my bad, I just saw the new test cases. I think that's a bug. Multiple groups with the same name should not be allowed. Anyway, here's an example of multiple captures:

```nim
let text = "hello world foo bar"
var m: RegexMatch
doAssert text.match(re"(?P<greet>hello) (?:(?P<who>[^\s]+)\s?)+", m)
# "who" captures @["world", "foo", "bar"]
doAssert m.groupCapture("who", text) == "world"
```

Reply:
I added that example test to the test suite and the docs

Reply:
Awesome, looks good. So, since we have this multi-capture feature, `groupFirstCapture`, `groupLastCapture` and `group` (returns seq[string]), they are all useful. I'm *not* asking you to add them, but rename `groupCapture` to `groupFirstCapture`, so the others can be added at some point.

Reply:
OK! added all of them please check if I missed anything and I updated the tests and they're green